### PR TITLE
Change Story in Editor

### PR DESCRIPTION
**Pull request by bhassan6621**
added a "story" button in editor that allows you to change the story rendered html, saves that edit in the state and updates the storyAd, so when you go back to the Ad's code in the html the changes you made to the story will appear there as well.

### CODE REVIEW

File: src/modify-storyAd.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {getNamespace} from '../lib/namespace';
+import {html} from 'lit-html';
+import {redispatchAs} from './utils/events';
+
+const {g} = getNamespace('modify-storyAd');
+
+// function cascadeInputClick({currentTarget}) {
```
Comment (alanorozco) [inline src/modify-storyAd.js]: Remove commented-out code? :)

File: src/modify-storyAd.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//   assert(currentTarget.getElementsByClassName('amp-sap_text-button')).click();
+// }
+
+const dispatchModifyStoryAd = redispatchAs(g('modify-storyAd'));
```
Comment (alanorozco) [inline src/modify-storyAd.js]: Let's rename this to "toggle inner outer" with all relevant units renamed to this as well.

File: src/modify-storyAd.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {getNamespace} from '../lib/namespace';
+import {html} from 'lit-html';
+import {redispatchAs} from './utils/events';
+
+const {g} = getNamespace('modify-storyAd');
```
Comment (alanorozco) [inline src/modify-storyAd.js]: `toggle-inner-outer`

File: src/modify-storyAd.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+// function cascadeInputClick({currentTarget}) {
+//   assert(currentTarget.getElementsByClassName('amp-sap_text-button')).click();
+// }
+
+const dispatchModifyStoryAd = redispatchAs(g('modify-storyAd'));
+
+export const ChangeDefaultStory = ({editingInner}) => html`
```
Comment (alanorozco) [inline src/modify-storyAd.js]: `ToggleInnerOuterContentButton`

File: src/editor.js [hunk 5]
```diff
@@ -116,6 +117,7 @@ const renderEditor = ({
   templates,
   templatesJson,
   viewportId = viewportIdDefault,
+  editingInner = true,
```
Comment (alanorozco) [inline src/editor.js]: Let's rename to `isEditingInner` for consistency.

**Review by alanorozco — DISMISSED** [review]

**Comment by calebcordry** [comment]
@bhassan6621 is this ready for review?

**Comment by bhassan6621** [comment]
> @bhassan6621 is this ready for review?

yes it's ready

File: src/story-patch.js [hunk 6]
```diff
@@ -15,7 +15,7 @@
  */
 import {minifyInlineJs} from './utils/minify-inline-js';
 
-export const cssPatch = minifyInlineJs(`<style amp-custom>
+export const cssPatch = minifyInlineJs(`
```
Comment (alanorozco) [inline src/story-patch.js]: Ah, don't `minifyInlineJs` this since it's `css`.

File: src/amp-story-ad-preview.js [hunk 7]
```diff
@@ -159,11 +179,26 @@ export default class AmpStoryAdPreview {
...
+    writeToIframe(await this.adIframe_, patch(dirty));
+  }
+
+  async updateOuter(dirty, dirtyInner) {
+    this.storyDoc = dirty;
+    this.adIframe_ = await awaitSelect(this.storyIframe_, 'iframe');
+    this.updateInnerInStory(dirtyInner);
+  }
+
+  async updateInnerInStory(dirty) {
```
Comment (calebcordry) [inline src/amp-story-ad-preview.js]: Should this method just be part of `updateOuter`?

**Review by calebcordry — CHANGES_REQUESTED** [review]
Looks good in general, just a couple of questions.

File: src/amp-story-ad-preview.js [hunk 8]
```diff
@@ -67,13 +68,24 @@ const httpsCircumventionPatch = minifyInlineJs(`
...
   docStr.replace(/<(body[^>]*)>/, '<$1 amp-story-visible>');
 
 const insertHttpsCircumventionPatch = docStr =>
-  docStr.replace('<head>', `<head><script>${httpsCircumventionPatch}</script>`);
+  addScriptToHead(docStr, httpsCircumventionPatch);
+
+const addScriptToHead = (docStr, headContent) =>
+  docStr.replace('<head>', `<head><script>${headContent}</script>`);
+
+const addCssToHead = (docStr, headContent) =>
```
[the remark this answers is not in the document]
Comment (bhassan6621) [reply src/amp-story-ad-preview.js]: yeah its pretty much the same, netlify didnt allow '<script>' tags inside minifyInlineJS() so i had to put the script tags in the addScriptToHead function, and I couldnt put the css inside the script tags so i made another const for adding the css

[the remark this answers is not in the document]
Comment (calebcordry) [reply src/amp-story-ad-preview.js]: You are right here :) I missed the style vs script.

File: src/amp-story-ad-preview.js [hunk 9]
```diff
@@ -67,13 +68,24 @@ const httpsCircumventionPatch = minifyInlineJs(`
...
+  docStr.replace('<head>', `<head><style>${headContent}</style>`);
+
+const storyNavigationPatch = (docStr, pageId) =>
+  addScriptToHead(docStr, navigationPatch.replace('$pageId$', pageId));
+
+const storyCssPatch = docStr => addCssToHead(docStr, cssPatch);
```
[the remark this answers is not in the document]
Comment (calebcordry) [reply src/amp-story-ad-preview.js]: lets just inline `addCssToHead` here

**Comment by calebcordry** [comment]
@bhassan6621 Are you still working on this PR?

File: src/amp-story-ad-preview.js [hunk 10]
```diff
@@ -159,11 +179,26 @@ export default class AmpStoryAdPreview {
...
    */
-  async update(dirty) {
+  async updateInner(dirty) {
     // TODO: Expose AMP runtime failures & either:
     // a) purifyHtml() from ampproject/src/purifier
     // b) reject when invalid
-    writeToIframe(await this.adIframe_, patch(dirty));
+    // Navigate back to ad page
     setMetaCtaLink(this.win, dirty, await this.storyCtaLink_);
+    this.adIframe_ = await awaitSelect(this.storyIframe_, 'iframe');
```
[the remark this answers is not in the document]
Comment (calebcordry) [reply src/amp-story-ad-preview.js]: Sorry maybe I missed something, why is this resolved?

**Review by calebcordry — APPROVED** [review]

**Comment by calebcordry** [comment]
@alanorozco merging this for now so that I can start on clean up work.

**Pull request merged**